### DISCUSSION TITLE
Specify only directory name for output

### DISCUSSION THREAD

**Discussion by skysurfer72** [DannyDainton/newman-reporter-htmlextra#391] [Q&A] [answered]
If I use the --reporter-htmlextra-export option I must specify a directory and a file name.
I like the autogenerated file name with date and time in it.
Is it possible to specify only output directory name?
I live in Italy and the time in the name of the file is GMT. Is it possible to specity timezone?

**Comment by DannyDainton** [comment, marked as answer]
That export option is baked into each custom reporter and comes from Newman, it's not something that I've individually added in htmlextra. If you don't use that flag it will push the report to the `/newman` directly, this will be created if you don't have one.

You could push the report with the default name to a different directory but **the directory would need to be created first**. 

```
newman run collection.postman_collection.json -r htmlextra --reporter-htmlextra-export ./reports
```
This would create the HTML report in the `./reports` directory, at the root level of the project you're running the command.

If you were to specify an export path with a new directory and give the HTML file a new name, `./reports/test_report.html`, it would create the new directory for you but this doesn't happen if you **only** use the directory in the command.


<img width="709" alt="image" src="https://user-images.githubusercontent.com/17089546/184321836-7cef36ce-84eb-47b8-87cb-6883b5169565.png">


For the second question:

<img width="967" alt="image" src="https://user-images.githubusercontent.com/17089546/184318872-46edc25f-b9d7-489a-af04-2b80e0ec657a.png">

Hope that's useful. 🏆

**Reply by skysurfer72** [reply]
Thank you Danny. My problem was that the destination folder didn't exist. Now, I manually created the folder and the file is saved where I want. 
 still got the problem with time in filename. I tried adding --reporter-htmlextra-timezone "Australia/Sydney" and the time changed in generated html file
![image](https://user-images.githubusercontent.com/12794564/184323530-905e42cd-adba-4397-8388-f43cd577f788.png)
But the file name is always Test-2022-08-12-09-11-28-312-0.html
But this is a very small problem.
Thanks

**Reply by DannyDainton** [reply]
The timezone flag is only used within the report to change the time but it doesn't impact that generated file name. It is not something that I'm going to be picking up - I'm more than happy to accept a PR for that feature if you'd like to give it a go. 🏆

**Reply by DannyDainton** [reply]
The timestamp is generated in the Newman code - https://github.com/postmanlabs/newman/blob/develop/lib/run/export-file.js#L68

It would be quite hacky on my reporters side to change that actually 🤔

**Reply by DannyDainton** [reply]
@skysurfer72 I remember having a similar discussion a couple of years ago and here's a possible solution for the file name.

https://github.com/DannyDainton/newman-reporter-htmlextra/issues/239#issuecomment-650506085

**Reply by skysurfer72** [reply]
Fantastic @DannyDainton ! Thank you so much!
